lib/credo_naming/check/warning/avoid_specific_terms_in_module_names.ex

```diff
@@ -56,6 +56,14 @@ defmodule CredoNaming.Check.Warning.AvoidSpecificTermsInModuleNames do
 
   def traverse(ast, issues, _, _), do: {ast, issues}
 
+  defp term_to_avoid?(term, terms) do
+    Enum.any?(terms, fn
+      term_to_avoid when is_binary(term_to_avoid) -> String.downcase(term_to_avoid) == String.downcase(term)
+      %Regex{} = term_to_avoid -> Regex.match?(term_to_avoid, term)
+      _ -> false
```

Comment:
It could be nice to raise on invalid config here

```suggestion
      term -> raise("The "terms" config expected each term to be a String or Regex, got: #{inspect(term)}")
```

The error format is inspired from Elixir source: https://github.com/elixir-lang/elixir/blob/fc039770b4f63eba4fa53acc077aeebed8603c3d/lib/elixir/lib/access.ex#L231